EOLib/Domain/Character/CharacterRepository.cs

```diff
@@ -22,5 +24,13 @@ public class CharacterRepository : ICharacterRepository, ICharacterProvider
         public bool HasAvatar { get; set; } = true;
 
         public Character MainCharacter { get; set; }
+
+        public bool InGuild
+        {
+            get
+            {
+                return MainCharacter.GuildTag != "   ";
```

Comment:
I think this property makes more sense as an auto property on `Character` rather than as part of the repository. For things like this use syntax `public bool InGuild => GuildTag != "   ";`

Also - is this the confirmed behavior of the official client?

Reply:
Can we put it on Character here? I thought it was part of eolib. It's not confirmed no just an assumption

Reply:
> Can we put it on Character here? I thought it was part of eolib.

I don't think I understand why this being part of eolib precludes adding the property there. Remember because of my dumbassery that `EOLib` in the context of dotnet is the shared library for EndlessClient where `Moffat.EndlessOnline.SDK` is what contains the generated code structures.

From a design perspective a character provider/repository should really only have a character in it. The repository isn't what owns the concept of a guild tag, the character is. If the character was in an unmodifiable library I'd still argue for an extension method on Character rather than putting a property on the repository.

Regarding `HasAvatar` - I genuinely don't remember why I did it this way but I don't like it 🤣 that should be refactored.

> It's not confirmed just an assumption

I'm assuming there's no better way to identify whether a player is part of a guild, then? Would the `GuildTag` ever be empty (`""` or `string.Empty`) or unset (`null`)?

Reply:
The server always sends 3 bytes no matter what. When not in a guild it's `   `

Reply:
I thought Character was part of the SDK for some reason. It's defined on `Character` now lmao